lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -2397,8 +2397,16 @@ struct FoldZeroWidthMemory : public mlir::RewritePattern {
     for (auto port : op->getResults()) {
       for (auto *user : llvm::make_early_inc_range(port.getUsers())) {
         SubfieldOp sfop = cast<SubfieldOp>(user);
-        replaceOpWithNewOpAndCopyName<WireOp>(rewriter, sfop,
-                                              sfop.getResult().getType());
+        auto wire = replaceOpWithNewOpAndCopyName<WireOp>(
+                        rewriter, sfop, sfop.getResult().getType())
+                        .getResult();
+        if (auto intType = type_dyn_cast<IntType>(wire.getType());
+            intType && intType.getBitWidthOrSentinel() == 0) {
+          // Make sure to write data ports.
+          auto zero = rewriter.create<firrtl::ConstantOp>(
+              wire.getLoc(), intType, APInt::getZero(0));
+          rewriter.create<StrictConnectOp>(wire.getLoc(), wire, zero);
+        }
```

Comment:
The second part of the `if` check `intType.getBitWidthOrSentinel() == 0` should be unnecessary given line 2386.

Also, under what circumstances would a blind `type_dyn_cast` _not_ work? Aggregate memory? If that's the case then maybe it makes more sense to not run this canonicalizer unless the memory has already been flattened.

Reply:
Added an early exit for aggregate memory.